realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -153,6 +188,37 @@ protected boolean funnelCollection(OsSet otherOsSet,
         }
     }
 
+    private boolean isObjectSameType(@Nullable Object o) {
+        // Return false when passing something else than the correct type
+        if (o != null) {
+            return o.getClass() == valueClass;
+        } else {
+            return true;
+        }
+    }
+
+    private boolean isUpperBoundCollectionSameType(Collection<? extends E> c) {
+        if (!c.isEmpty()) {
+            for (E item : c) {
+                if (item != null) {
+                    return item.getClass() == valueClass;
```

Comment:
You should only return if they are not equals, otherwise you would have to continue.

Reply:
Oh I had it like you mention but got a warning from AS. This is what I got after the IDE suggested I should change it because I was performing a "negative check" or something like that and didn't really double-checked how it looked after the change. I will revert it.